### ISSUE TITLE
Authenticate to mongodb

### ISSUE THREAD

**Issue by kevb** [kongo2002/ejabberd-mod-mam#9]
Is there any option to authenticate to the mongodb. My mongodb is running on another host and I need to be able to authenticate the connection. Thanks

**Comment by kongo2002** [comment]
Hi,
I am kind of shocked because I was pretty sure it's possible but you are right indeed. I'll see what I can do.

Cheers,
Gregor

**Comment by kevin-bennett-ags** [comment]
Thanks Gregor, it would be very much appreciated.

I noted that the new mongodb driver (https://github.com/comtihon/mongodb-erlang) has support for authentication but the interface has changed quite a bit vs the one you're using.

I wish I could help but I'm not confident enough in erlang.

Cheers

**Comment by kongo2002** [comment]
Yes, the 'new' mongodb driver was a huge rewrite while dropping support for replica sets and others for example. That's why I still kept the old one. I haven't checked lately but maybe the new one matured in the meantime.

**Comment by kevin-bennett-ags** [comment]
Yeah, I noticed that a while back. But I believe it now supports replica sets and has matured a lot.

I have my fingers crossed that the new driver is OK and you find time to add support. Thanks!

**Comment by kongo2002** [comment]
I think I will try to add authentication support for the old mongodb driver first and make the migration to the new driver a separate issue.
